src/io.jl

```diff
@@ -127,6 +129,33 @@ function _loadtable(T, files::Union{AbstractVector,String};
         filegroups = filter(!isempty, map(x->files[x], chunks))
     end
 
+    if blocksize > 0
+        # Use blocked reader
+        blockgroups = []
+        for _files in filegroups
+            for file in _files
+                # Break file into blocks of size `blocksize` or less
+                fsize = filesize(file)
+                nblocks = max(div(fsize, blocksize), 1)
+                bios = blocks(file, '\n', nblocks)
+                # Extract header
+                # FIXME: Use csvread instead!
+                header_str = open(file, "r") do hdrio
+                    readline(hdrio)
+                end
+                header = replace.(string.(split(header_str, ',')),
+                                  Ref(r"\s" => ""))
+                for (idx,block) in enumerate(bios)
+                    block.l > 0 && push!(blockgroups, (file, idx, header))
```

Comment:
It seems pretty expensive to read all the files char-by-char on master just to make sure no block is non-empty, and then ask the workers to read it again. Why not just send the workers the number of blocks and the index? (this is even slightly better than sending offset and length like I suggested, because those can be computed from nblocks and idx without extra work than the worker already does now).

Reply:
It's pretty cheap to just pass the range as-is (given that we're passing the whole header too), so I'll send that. It should make constructing a new `BlockIO` easy.